### PR TITLE
Add user-specific settings page

### PR DESCRIPTION
**Pull request by ahornace**
Screen Shot 2021-06-19 at 12 31 45

This is an attempt to introduce user settings page which would allow every user to slightly modify their OpenGrok experience in a nicer way (i.e. not using cookies). It uses local storage to store the values which should be persisted for a long time unless user explicitly clears them.

This change just introduces suggester enabled/disabled checkbox and is meant to be a founding stone for more things like:
- suggester configuration (show projects, speed time, number of results, min chars, ...)
- font type or size
- line spacing
- color themes
- default project
- ...

https://github.com/oracle/opengrok/wiki/Cookies will need to be modified.

Tested in Chrome and Safari.

Thanks!

### CODE REVIEW

File: opengrok-web/src/main/webapp/js/utils-0.0.41.js [hunk 1]
```diff
@@ -1700,8 +1700,7 @@ function pageReadyMast() {
 }
 
 function domReadyMenu(minisearch) {
-    if (getCookie('OpenGrokSuggester.enabled') === 'false') {
-        console.log('Suggester disabled by a cookie');
```
Comment (vladak) [inline opengrok-web/src/main/webapp/js/utils-0.0.41.js:1704]: do we want to retain some sort of `console.log()` ?

File: opengrok-web/src/main/webapp/js/utils-0.0.41.js [hunk 2]
```diff
@@ -2378,22 +2377,76 @@ function escapeHtml(string) { // taken from https://stackoverflow.com/questions/
...
+}
+
+/**
+ * Initializes all settings elements with {@code local-setting} class.
+ */
+function initSettings() {
+    for (const el of document.getElementsByClassName('local-setting')) {
+        const value = getSettingsValue(el.name);
+        if (value) {
+            if (el.type === 'checkbox') {
```
Comment (vladak) [inline opengrok-web/src/main/webapp/js/utils-0.0.41.js:2443]: could be refactored w.r.t. setDefaultSettingsValue() ?

**Review by vladak — APPROVED** [review]

File: opengrok-web/src/main/webapp/js/utils-0.0.41.js (hunk 1 above)
Comment (ahornace) [reply opengrok-web/src/main/webapp/js/utils-0.0.41.js:1704]: I think it's a good javascript practice to remove `console.log()` calls from production code. I quickly checked if google closure could not be configured to do that but it does not seem so. Although this hack could be used: https://stackoverflow.com/questions/2934509/exclude-debug-javascript-code-during-minification#answer-5969752

But I can add it back if you think the logs are beneficial as I do not have a strong opinion about it.

Comment (vladak) [reply opengrok-web/src/main/webapp/js/utils-0.0.41.js:1704]: I am fine with whatever the current best practice is.

**Pull request merged**